### ISSUE TITLE
REP4: Requirements for DNS

### ISSUE THREAD

**Issue by zefciu** [allegro/ralph#1828]
The following features should be available in Ralph NG in order to support integration with DNSaaS service by https://github.com/allegro/django-powerdns-dnssec
1. A database of all the domains managed by the company (not only the ones that we have contract for, but also those served by our DNS servers) together with service-environment and technical owners of these domains
2. Ability to push domain data via API to Ralph
3. Pushing changes done to existing domains in Ralph.
4. Ability to propopulate the data about already existing domains from CSV or similar format.

The agreed upon workflow will look like this:
- We will try to guess as much as possible about existing domains based on the services' URLs and owners
- For the rest of existing domains we will gather the information manually by contacting services' owners
- New domains will be created via DNSaaS and pushed to Ralph immediately
- Changes done in Ralph will be pushed immediately to DNSaaS

This way we don't have any need for scheduled jobs and the data should be constantly synchronised between Ralph and DNSaaS.

**Comment by mkurek** [comment]
I disagree with `Changes done in Ralph will be pushed immediately to DNSaaS` - IMO Ralph shouldn't have any idea that there is something called DNSaaS - in every case DNSaaS should either pull or push data to Ralph.

But, if we (Ralph) provide pub-sub mechanism (for example using Redis), then Ralph could publish info that domain has changed, and DNSaaS will subscribe to this type of message.

**Comment by vi4m** [comment]
+1 for systems agnostic solution - without updating dependent subsystems like DNSaaS. pub/sub mechanism looks like good idea.

**Comment by vi4m** [comment]
From the Assets management point of view we have:

``` python
class Domain(BaseObject):
    name = models.CharField(
        verbose_name=_('Domain name'),
        help_text=_('Full domain name'),
        unique=True,
        max_length=255
    )
    domain_status = models.PositiveIntegerField(
        choices=DomainStatus(),
        default=DomainStatus.active.id,
    )
    billing_company = models.CharField(
        max_length=255, blank=True,
        help_text=_("Company which receives invoice for domain")
    )
    business_segment = models.ForeignKey(
        BusinessSegment, blank=True, null=True,
        help_text=_("Business segment for a domain")
    )
    business_owner = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        related_name='domaincontract_business_owner',
        blank=True,
        null=True,
        help_text=_("Business owner of a service")
    )
    technical_owner = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        related_name='domaincontract_technical_owner',
        blank=True,
        null=True,
        help_text=_("Technical owner of a service")
    )
    domain_holder = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        related_name='domaincontract_domain_holder',
        blank=True,
        null=True,
        help_text=_("User which is directly responsible for the domain renewal")  # noqa
    )
```

**Comment by vi4m** [comment]
So the "junction" point will be the `Domain` object, between Assets/detailed dns records. Nowe we have to discuss how Ralph will interact with the DNSaas service.

**Comment by mkurek** [comment]
Implemented in #2289
